### ISSUE TITLE
Some weights of BlipModel were not initialized from the model checkpoint at Salesforce/blip-image-captioning-base.

### ISSUE THREAD

**Issue by aivolcano** [huggingface/transformers#31199]
### System Info

Some weights of BlipModel were not initialized from the model checkpoint at Salesforce/blip-image-captioning-base. This bug is thrown when I run the offical code in https://huggingface.co/docs/transformers/v4.41.3/en/model_doc/blip#transformers.BlipModel
![截图 2024-06-03 16-03-57](https://github.com/huggingface/transformers/assets/68730894/6be49972-c218-4669-8120-7aae4c82d9ba)

below is the offical code. 
```python
from PIL import Image
import requests
from transformers import AutoProcessor, BlipModel

model = BlipModel.from_pretrained("Salesforce/blip-image-captioning-base")
processor = AutoProcessor.from_pretrained("Salesforce/blip-image-captioning-base")

url = "http://images.cocodataset.org/val2017/000000039769.jpg"
image = Image.open(requests.get(url, stream=True).raw)

inputs = processor(
    text=["a photo of a cat", "a photo of a dog"], images=image, return_tensors="pt", padding=True
)

outputs = model(**inputs)
logits_per_image = outputs.logits_per_image  # this is the image-text similarity score
probs = logits_per_image.softmax(dim=1)  # we can take the softmax to get the label probabilities
``` 
 @ArthurZucker and @younesbelkada @amyeroberts

### Who can help?

 @ArthurZucker and @younesbelkada @amyeroberts

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

from PIL import Image
import requests
from transformers import AutoProcessor, BlipModel

model = BlipModel.from_pretrained("Salesforce/blip-image-captioning-base")
processor = AutoProcessor.from_pretrained("Salesforce/blip-image-captioning-base")

url = "http://images.cocodataset.org/val2017/000000039769.jpg"
image = Image.open(requests.get(url, stream=True).raw)

inputs = processor(
    text=["a photo of a cat", "a photo of a dog"], images=image, return_tensors="pt", padding=True
)

outputs = model(**inputs)
logits_per_image = outputs.logits_per_image  # this is the image-text similarity score
probs = logits_per_image.softmax(dim=1)  # we can take the softmax to get the label probabilities

### Expected behavior

The model weight will be initialized correctly when running the official code

**Comment by RUFFY-369** [comment]
Hi @aivolcano , it is a known issue with `BlipModel` because this class isn't compatible right now with weights that are on the HF hub and there are already two issues closed regarding this. Please have a look at them as these contain the solutions for your issue as well. Here they are: #28034  #25024 
And I think this class will be removed from the docs and deprecated.

Cheers!
